### PR TITLE
ignore node_modules directory while compiling python files

### PR DESCRIPTION
**Pull request by ziafazal**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by ziafazal** [comment]
@fredsmith this PR changes the python compileall regex to exclude node_modules directory also while compiling python files. Could you please review?

**Comment by jibsheet** [comment]
@ziafazal This is failing tests, and the failure looks legitimate.
Do you know why your fork was installing an upgraded node module rather than something pinned?

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 1]
```diff
@@ -300,7 +300,7 @@
     - install:app-requirements
 
 - name: compiling all py files in the edx-platform repo
-  shell: "{{ edxapp_venv_bin }}/python -m compileall -q -x .git/.* {{ edxapp_code_dir }}"
+  shell: "{{ edxapp_venv_bin }}/python -m compileall -q -x .git/.*|node_modules/.* {{ edxapp_code_dir }}"
```
Comment (fredsmith) [inline playbooks/roles/edxapp/tasks/deploy.yml]: this is a shell command - why are you piping things?  What are you trying to do here?

**Comment by ziafazal** [comment]
@jibsheet I don't have any idea what has caused installing an upgraded node module. I have been told this [xblock](https://github.com/open-craft/problem-builder/commits/master) might have caused it but I'm not sure.

File: playbooks/roles/edxapp/tasks/deploy.yml (hunk 1 above)
Comment (fredsmith) [inline playbooks/roles/edxapp/tasks/deploy.yml]: This should be single-quoted, so doesn't get parsed by the shell.

**Comment by jibsheet** [comment]
@nedbat FYI that this will impact Ginkgo and Ficus (someone in #ops reported it on ficus) because edx-ui-toolkit doesn't pin sinon well and newer sinons eventually pull in js-yaml which has a python3 file in it under node_modules.

**Comment by ziafazal** [comment]
@fredsmith build passed after adding quotes to regex. Let me squash commits.

**Review by fredsmith — APPROVED** [review]
This looks good to me.

**Review by manikarthikk — APPROVED** [review]

**Comment by manikarthikk** [comment]
I verified this with a deployment and this looks good.

**Pull request merged**